components/fd/component_output.go

```diff
@@ -116,37 +149,54 @@ func (o *Output) States() ([]components.State, error) {
 		Name:    StateNameFileDescriptors,
 		Healthy: true,
-		Reason:  fmt.Sprintf("running_pids: %d, usage: %d, limit: %d, threshold_limit: %d, used_percent: %s, threshold_used_percent: %s", o.RunningPIDs, o.Usage, o.Limit, o.ThresholdLimit, o.UsedPercent, o.ThresholdUsedPercent),
+		Reason: fmt.Sprintf("allocated_file_handles: %d, allocated_percent: %s, running_pids: %d, used_percent: %s",
+			o.AllocatedFileHandles,
+			o.AllocatedFileHandlesPercent,
+			o.RunningPIDs,
+			o.UsedPercent,
+		),
 		ExtraInfo: map[string]string{
-			StateKeyRunningPIDs: fmt.Sprintf("%d", o.RunningPIDs),
-			StateKeyUsage:       fmt.Sprintf("%d", o.Usage),
+			StateKeyAllocatedFileHandles: fmt.Sprintf("%d", o.AllocatedFileHandles),
+			StateKeyRunningPIDs:          fmt.Sprintf("%d", o.RunningPIDs),
+			StateKeyUsage:                fmt.Sprintf("%d", o.Usage),
+			StateKeyLimit:                fmt.Sprintf("%d", o.Limit),
 
-			StateKeyLimit:       fmt.Sprintf("%d", o.Limit),
-			StateKeyUsedPercent: o.UsedPercent,
+			StateKeyAllocatedFileHandlesPercent: o.AllocatedFileHandlesPercent,
+			StateKeyUsedPercent:                 o.UsedPercent,
 
-			StateKeyFDLimitSupported: fmt.Sprintf("%v", o.FDLimitSupported),
+			StateKeyThresholdAllocatedFileHandles:        fmt.Sprintf("%d", o.ThresholdAllocatedFileHandles),
+			StateKeyThresholdAllocatedFileHandlesPercent: o.ThresholdAllocatedFileHandlesPercent,
 
-			StateKeyThresholdLimit:       fmt.Sprintf("%d", o.ThresholdLimit),
-			StateKeyThresholdUsedPercent: o.ThresholdUsedPercent,
+			StateKeyThresholdRunningPIDs:        fmt.Sprintf("%d", o.ThresholdRunningPIDs),
+			StateKeyThresholdRunningPIDsPercent: o.ThresholdRunningPIDsPercent,
+
+			StateKeyFileHandlesSupported: fmt.Sprintf("%v", o.FileHandlesSupported),
+			StateKeyFDLimitSupported:     fmt.Sprintf("%v", o.FDLimitSupported),
 		},
 	}
 
+	if allocatedPercent, err := o.GetAllocatedFileHandlesPercent(); err == nil && allocatedPercent > 95.0 {
+		state.Healthy = false
+		state.Reason += "; allocated_file_handles_percent is greater than 95"
+	}
 	if usedPercent, err := o.GetUsedPercent(); err == nil && usedPercent > 95.0 {
 		state.Healthy = false
-		state.Reason += "-- used_percent is greater than 95"
+		state.Reason += "; used_percent is greater than 95"
 	}
 
-	if o.FDLimitSupported && o.ThresholdLimit > 0 {
-		if thresholdUsedPercent, err := o.GetThresholdUsedPercent(); err == nil && thresholdUsedPercent > 95.0 {
-			state.Healthy = false
-			state.Reason += "-- threshold_used_percent is greater than 95"
-		}
+	if o.FDLimitSupported && o.ThresholdRunningPIDs > 0 && o.RunningPIDs > o.ThresholdRunningPIDs {
+		state.Healthy = false
+		state.Reason += "; running_pids is greater than threshold_running_pids"
+	}
+	if o.FileHandlesSupported && o.ThresholdAllocatedFileHandles > 0 && o.AllocatedFileHandles > o.ThresholdAllocatedFileHandles {
+		state.Healthy = false
+		state.Reason += "; allocated_file_handles is greater than threshold_allocated_file_handles"
 	}
 
```

Comment:
here maybe we should check ThresholdAllocatedFileHandlesPercent as well (also give a default value like 80%), imo ThresholdAllocatedFileHandlesPercent is more important than the absolute value of fd count..